From f1542ae9fe10d5a241fc2624ecaef5f0948e3472 Mon Sep 17 00:00:00 2001
From: Michael Matz <matz@suse.de>
Date: Mon, 25 Mar 2013 01:06:56 +0100
Subject: [PATCH 123/169] Don't block SIGSEGV at more places.

This is used internally by qemu to detect writing to pages
containing code, it must not be disabled even when the guest
disables these.
---
 linux-user/signal.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: qemu-1.6.0+dfsg/linux-user/signal.c
===================================================================
--- qemu-1.6.0+dfsg.orig/linux-user/signal.c	2013-11-05 22:23:41.179369355 +0000
+++ qemu-1.6.0+dfsg/linux-user/signal.c	2013-11-05 22:23:41.171369375 +0000
@@ -1259,7 +1259,7 @@
     uint32_t magic, size;
 
     target_to_host_sigset(&set, &sf->uc.tuc_sigmask);
-    sigprocmask(SIG_SETMASK, &set, NULL);
+    do_sigprocmask(SIG_SETMASK, &set, NULL);
 
     for (i = 0; i < 31; i++) {
         err |= __get_user(env->xregs[i], &sf->uc.tuc_mcontext.regs[i]);
@@ -5798,6 +5798,7 @@
             sigaddset(&set, target_to_host_signal(sig));
 
         /* block signals in the handler using Linux */
+        sigdelset(&set, SIGSEGV);
         sigprocmask(SIG_BLOCK, &set, &old_set);
         /* save the previous blocked signal state to restore it at the
            end of the signal execution (see do_sigreturn) */
